File: webapp/src/types/ResponseMessage.ts
export interface ResponseMessage<T> {
	status: string;
	message: string;
	data: T | null;
}

// ----- type guards -----

export function isResponseMessage<T>(obj: any, isDataType: (data: any) => data is T): obj is ResponseMessage<T> {
	return (
		obj &&
		obj.hasOwnProperty('status') &&
		typeof obj.status === 'string' &&
		obj.hasOwnProperty('message') &&
		typeof obj.message === 'string' &&
		obj.hasOwnProperty('data') &&
		(obj.data === null || isDataType(obj.data))
	);
}

export function isResponseMessageSuccess(responseMessage: ResponseMessage<any>) {
	return responseMessage.status === 'success';
}

File: backend/giffyLambda/lambda/src/main/java/com/adventuroushachi/function/ResponseMessage.java
package com.adventuroushachi.function;

public class ResponseMessage<T> {
    private final String status;
    private final String message;
    private final T data;

    public ResponseMessage(ResponseMessageStatus status, String message, T data) {
        this.status = status.getStatus();
        this.message = message;
        this.data = data;
    }

    public String getStatus() {
        return status;
    }

    public String getMessage() {
        return message;
    }

    public T getData() {
        return data;
    }
}
